### ISSUE TITLE
Test::TCP fails to install on Windows 7 64-bit

### ISSUE THREAD

**Issue by azawawi** [tokuhirom/Test-TCP#43]
Test::TCP fails to install on Windows 7 Ultimate 64-bit on perl v5.22.0

```
cpanm (App::cpanminus) 1.7036 on perl 5.022000 built for MSWin32-x64-multi-thread
Work directory is C:\Users\azawawi/.cpanm/work/1435228423.6632
You have make C:\Strawberry\c\bin\dmake.exe
You have LWP 6.13
Falling back to Archive::Tar 2.04
Searching Test::TCP () on cpanmetadb ...
--> Working on Test::TCP
Fetching http://www.cpan.org/authors/id/T/TO/TOKUHIROM/Test-TCP-2.12.tar.gz
-> OK
Unpacking Test-TCP-2.12.tar.gz
Entering Test-TCP-2.12
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.04_01)
Configuring Test-TCP-2.12
Running Makefile.PL
Checking if your kit is complete...
Looks good
Generating a dmake-style Makefile
Writing Makefile for Test::TCP
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have IO::Socket::IP 0 ... Yes (0.37)
Checking if you have Test::More 0.98 ... Yes (1.001014)
Checking if you have Test::SharedFork 0.29 ... Yes (0.33)
Checking if you have File::Temp 0 ... Yes (0.2304)
Checking if you have Time::HiRes 0 ... Yes (1.9726)
Checking if you have Socket 0 ... Yes (2.020)
Checking if you have IO::Socket::INET 0 ... Yes (1.35)
Building and testing Test-TCP-2.12
cp lib/Test/TCP.pm blib\lib\Test\TCP.pm
cp lib/Net/EmptyPort.pm blib\lib\Net\EmptyPort.pm
cp lib/Test/TCP/CheckPort.pm blib\lib\Test\TCP\CheckPort.pm
"C:\Strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
# Test::More: 1.001014
t/00_compile.t ................. ok
cannot open port: ::1:51417 at C:\Users\azawawi\.cpanm\work\1435228423.6632\Test-TCP-2.12\blib\lib/Test/TCP.pm line 54.
    # Child (v6) exited without calling finalize()

#   Failed test 'v6'
#   at C:/Strawberry/perl/lib/Test/Builder.pm line 279.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 2.
t/01_simple.t .................. 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/2 subtests 
t/02_abrt.t .................... skipped: win32 doesn't support embedded function named dump()
t/03_return_when_sigterm.t ..... ok
t/04_die.t ..................... ok
t/05_sigint.t .................. skipped: this test requires SIGUSR1
t/06_nest.t .................... ok
t/07_optional.t ................ ok
t/08_exit.t .................... ok
t/09_fork.t .................... ok
t/10_oo.t ...................... ok

    #   Failed test 'port is open'
    #   at t/11_net_empty_port.t line 22.
    # Looks like you failed 1 test of 4.

#   Failed test 'v6'
#   at t/11_net_empty_port.t line 37.
# Looks like you failed 1 test of 4.
t/11_net_empty_port.t .......... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/4 subtests 
t/12_pass_wait_port_options.t .. ok
t/13_undef_port.t .............. ok

Test Summary Report
-------------------
t/01_simple.t                (Wstat: 65280 Tests: 2 Failed: 1)
  Failed test:  2
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
t/11_net_empty_port.t        (Wstat: 256 Tests: 4 Failed: 1)
  Failed test:  4
  Non-zero exit status: 1
Files=14, Tests=117, 86 wallclock secs ( 0.11 usr +  0.00 sys =  0.11 CPU)
Result: FAIL
Failed 2/14 test programs. 2/117 subtests failed.
dmake.exe:  Error code 255, while making 'test_dynamic'
-> FAIL Installing Test::TCP failed. See C:\Users\azawawi\.cpanm\work\1435228423.6632\build.log for details. Retry with --force to force install it.
```

**Comment by mattn** [comment]
related on #42 ?

**Comment by azawawi** [comment]
Tried #42 but still there are errors. I also tried the following but it did not work out:

* Disable the firewall
* Run in administrator mode (no UAC)
* Disable anti-virus

**Comment by wchristian** [comment]
Please try again with the newest version so we can see the remaining issues. :)

**Comment by azawawi** [comment]
Still fails. Different errors :)

```
cpanm (App::cpanminus) 1.7036 on perl 5.022000 built for MSWin32-x64-multi-thread
Work directory is C:\Users\azawawi/.cpanm/work/1437898143.4780
You have make C:\Strawberry\c\bin\dmake.exe
You have LWP 6.13
Falling back to Archive::Tar 2.04
Searching Test::TCP () on cpanmetadb ...
--> Working on Test::TCP
Fetching http://www.cpan.org/authors/id/T/TO/TOKUHIROM/Test-TCP-2.13.tar.gz
-> OK
Unpacking Test-TCP-2.13.tar.gz
Entering Test-TCP-2.13
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.64 ... Yes (7.04_01)
Configuring Test-TCP-2.13
Running Makefile.PL
Checking if your kit is complete...
Looks good
Generating a dmake-style Makefile
Writing Makefile for Test::TCP
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Time::HiRes 0 ... Yes (1.9726)
Checking if you have IO::Socket::INET 0 ... Yes (1.35)
Checking if you have Test::SharedFork 0.29 ... Yes (0.33)
Checking if you have File::Temp 0 ... Yes (0.2304)
Checking if you have IO::Socket::IP 0 ... Yes (0.37)
Checking if you have Socket 0 ... Yes (2.020)
Checking if you have Test::More 0.98 ... Yes (1.001014)
Building and testing Test-TCP-2.13
cp lib/Net/EmptyPort.pm blib\lib\Net\EmptyPort.pm
cp lib/Test/TCP/CheckPort.pm blib\lib\Test\TCP\CheckPort.pm
cp lib/Test/TCP.pm blib\lib\Test\TCP.pm
"C:\Strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
# Test::More: 1.001014
t/00_compile.t ................. ok
cannot open port: ::1:51006 at C:\Users\azawawi\.cpanm\work\1437898143.4780\Test-TCP-2.13\blib\lib/Test/TCP.pm line 54.
    # Child (v6) exited without calling finalize()

#   Failed test 'v6'
#   at C:/Strawberry/perl/lib/Test/Builder.pm line 279.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 2.
t/01_simple.t .................. 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/2 subtests 
t/02_abrt.t .................... skipped: win32 doesn't support embedded function named dump()
t/03_return_when_sigterm.t ..... ok
t/04_die.t ..................... ok
t/05_sigint.t .................. skipped: this test requires SIGUSR1
t/06_nest.t .................... ok
t/07_optional.t ................ ok
t/08_exit.t .................... ok
t/09_fork.t .................... ok
t/10_oo.t ...................... ok
    
    #   Failed test 'port is open'
    #   at t/11_net_empty_port.t line 22.
    # Looks like you failed 1 test of 4.

#   Failed test 'v6'
#   at t/11_net_empty_port.t line 37.
# Looks like you failed 1 test of 4.
t/11_net_empty_port.t .......... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/4 subtests 
t/12_pass_wait_port_options.t .. ok
t/13_undef_port.t .............. ok

Test Summary Report
-------------------
t/01_simple.t                (Wstat: 65280 Tests: 2 Failed: 1)
  Failed test:  2
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
t/11_net_empty_port.t        (Wstat: 256 Tests: 4 Failed: 1)
  Failed test:  4
  Non-zero exit status: 1
Files=14, Tests=117, 86 wallclock secs ( 0.05 usr +  0.03 sys =  0.08 CPU)
Result: FAIL
Failed 2/14 test programs. 2/117 subtests failed.
dmake.exe:  Error code 255, while making 'test_dynamic'
-> FAIL Installing Test::TCP failed. See C:\Users\azawawi\.cpanm\work\1437898143.4780\build.log for details. Retry with --force to force install it.
```

**Comment by paultcochrane** [comment]
I too am getting these errors, however on Strawberry Perl 5.20.3.1 (64 bit) on Windows 7 Professional (see below).  Nevertheless, the distribution installs without a problem on Strawberry Perl 5.18.4.1 (64 bit).

    cpanm (App::cpanminus) 1.7039 on perl 5.020003 built for MSWin32-x64-multi-thread
    Work directory is C:\Users\PAULCO~1/.cpanm/work/1445358606.1944
    You have make C:\Strawberry\c\bin\dmake.exe
    You have LWP 6.13
    Falling back to Archive::Tar 2.04
    Searching Test::TCP () on cpanmetadb ...
    --> Working on Test::TCP
    Fetching http://www.cpan.org/authors/id/T/TO/TOKUHIROM/Test-TCP-2.14.tar.gz
    -> OK
    Unpacking Test-TCP-2.14.tar.gz
    Entering Test-TCP-2.14
    Checking configure dependencies from META.json
    Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.10)
    Configuring Test-TCP-2.14
    Running Makefile.PL
    Checking if your kit is complete...
    Looks good
    Generating a dmake-style Makefile
    Writing Makefile for Test::TCP
    Writing MYMETA.yml and MYMETA.json
    -> OK
    Checking dependencies from MYMETA.json ...
    Checking if you have File::Temp 0 ... Yes (0.2304)
    Checking if you have Time::HiRes 0 ... Yes (1.9726)
    Checking if you have IO::Socket::IP 0 ... Yes (0.37)
    Checking if you have Socket 0 ... Yes (2.020)
    Checking if you have Test::SharedFork 0.29 ... Yes (0.34)
    Checking if you have Test::More 0.98 ... Yes (1.001014)
    Checking if you have IO::Socket::INET 0 ... Yes (1.35)
    Building and testing Test-TCP-2.14
    cp lib/Test/TCP/CheckPort.pm blib\lib\Test\TCP\CheckPort.pm
    cp lib/Test/TCP.pm blib\lib\Test\TCP.pm
    cp lib/Net/EmptyPort.pm blib\lib\Net\EmptyPort.pm
    "C:\Strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
    # Test::More: 1.001014
    t/00_compile.t ................. ok
    cannot open port: ::1:50052 at C:\Users\PAULCO~1\.cpanm\work\1445358606.1944\Test-TCP-2.14\blib\lib/Test/TCP.pm line 54.
	# Child (v6) exited without calling finalize()

    #   Failed test 'v6'
    #   at C:/Strawberry/perl/lib/Test/Builder.pm line 279.
    # Tests were run but no plan was declared and done_testing() was not seen.
    # Looks like your test exited with 255 just after 2.
    t/01_simple.t .................. 
    Dubious, test returned 255 (wstat 65280, 0xff00)
    Failed 1/2 subtests 
    t/02_abrt.t .................... skipped: win32 doesn't support embedded function named dump()
    t/03_return_when_sigterm.t ..... ok
    t/04_die.t ..................... ok
    t/05_sigint.t .................. skipped: this test requires SIGUSR1
    t/06_nest.t .................... ok
    t/07_optional.t ................ ok
    t/08_exit.t .................... ok
    t/09_fork.t .................... ok
    t/10_oo.t ...................... ok
	
	#   Failed test 'port is open'
	#   at t/11_net_empty_port.t line 22.
	# Looks like you failed 1 test of 4.

    #   Failed test 'v6'
    #   at t/11_net_empty_port.t line 37.
    # Looks like you failed 1 test of 4.
    t/11_net_empty_port.t .......... 
    Dubious, test returned 1 (wstat 256, 0x100)
    Failed 1/4 subtests 
    t/12_pass_wait_port_options.t .. ok
    t/13_undef_port.t .............. ok

    Test Summary Report
    -------------------
    t/01_simple.t                (Wstat: 65280 Tests: 2 Failed: 1)
      Failed test:  2
      Non-zero exit status: 255
      Parse errors: No plan found in TAP output
    t/11_net_empty_port.t        (Wstat: 256 Tests: 4 Failed: 1)
      Failed test:  4
      Non-zero exit status: 1
    Files=14, Tests=117, 90 wallclock secs ( 0.06 usr +  0.03 sys =  0.09 CPU)
    Result: FAIL
    Failed 2/14 test programs. 2/117 subtests failed.
    dmake.exe:  Error code 255, while making 'test_dynamic'
    -> FAIL Installing Test::TCP failed. See C:\Users\PAULCO~1\.cpanm\work\1445358606.1944\build.log for details. Retry with --force to force install it.

**Comment by sam0737** [comment]
Same here. On Windows 10 x64 with Strawberry Perl v5.20.1 x86. No antivirus, with default UAC and default firewall settings.

**Comment by itcharlie** [comment]
Hello,
Just wanted to note that I also have the same error with perl 5.22.0 on Win 7 64 bit installation but this error is not present in Windows for  Perl version 5.22.1 and 5.23.5  as you can see from the cpantester matrix : http://matrix.cpantesters.org/?dist=Test-TCP%202.14;os=mswin32;reports=1#sl=4,1

**Comment by richardleach** [comment]
I had the same problem with Strawberry 5.22.1 on 64-bit Win 10. It traced back to a strange bit of Windows-centric code in Net::EmptyPort::wait_port, specifically line 113:

`if ($^O eq 'MSWin32' ? `$^X -MTest::TCP::CheckPort -echeck_port $port $proto` : check_port({ host => $host, port => $port, proto => $proto })) {`

When running the IPv6 tests on MSWin32 platforms, the failure to pass the $host to the command in backticks means that Net::EmptyPort::check_port tries to check for the open port on '127.0.0.1', not on '::1' - and then inevitably fails to find the open port.

I patched the Windows specific code out, so that check_port() was called in the same way regardless of platform and Test::TCP then built without problem:

`if ( check_port({ host => $host, port => $port, proto => $proto }) ) {`

**Comment by chorny** [comment]
Both 32-bit and 64-bit perl have this problem on 64-bit Windows 7, but not on 32-bit Windows 7.
Solution is to install version 2.07.

**Comment by zdm** [comment]
I can confirm, that Test::TCP never installs on any Windows for me on. 
This package is a dependency for Plack and break automated deployment.

**Comment by miyagawa** [comment]
> This package is a dependency for Plack and break automated deployment.

You can skip running tests with `--notest` and that should make your automated deployment work. We're aware of the issue but do not try to make it more dramatic than what it is.

**Comment by zdm** [comment]
Yes, I know, but I prefer to run tests to make sure, that everything is OK.

**Comment by zdm** [comment]
Maybe, if those tests can't be passed on Windows, they should be disabled for Windows?

**Comment by charsbar** [comment]
They can, and they do pass for me (Win7 + Win32 strawberry-perl 5.20.0.1 #1 Fri May 30 08:27:00 2014 x64).

**Comment by rzl** [comment]
This is perl 5, version 22, subversion 0 (v5.22.0) built for MSWin32-x64-multi-thread
win7 64-bit it still a problem ?  i install Redis has the same error message .  
skip running tests Test::TCP and reinstall Redis it will success.
thanks all big god .

**Comment by nponeccop** [comment]
Can you retest against https://github.com/tokuhirom/Test-TCP/pull/67 ?

**Comment by paultcochrane** [comment]
I've retested using Strawberry Perl 5.24.0 and 5.20.3.3 on Windows 7 and was able to successfully install version 2.19 of `Test::TCP` (via `cpanm`).  Both versions of Perl also test correctly against the current state of the `master` branch (fb10bdf7a64f79e387042cca568f481c37c3172d).  It would seem that sometime between version 2.14 and 2.19 that this issue has been resolved.

**Comment by nponeccop** [comment]
> Both versions of Perl also test correctly against the current state of the master branch (fb10bdf).

Can you check that there's no newer Test::Simple in StrawberryPerl? The current branch passes the tests as long as you have a newer Test::Simple installed. Te goal of #67 is to force the update of Test::Simple on clean installations.

**Comment by paultcochrane** [comment]
The test with SP 5.20.3.3 was with a clean install, so I'm guessing the default `Test::Simple` was used there (I didn't keep build logs for tests with this version).  I've just tried fb10bdf again with a clean install of SP 5.20.3.1 (the version I used in an earlier comment) and the tests pass.  The version of `Test::More` was also the same as that used in my first comment in this thread (1.001014).  Here's the test output:

```
C:\Users\cochrane\Test-TCP>cpanm --installdeps .
--> Working on .
Configuring Test-TCP-2.19 ... OK
==> Found dependencies: Test::SharedFork
--> Working on Test::SharedFork
Fetching http://www.cpan.org/authors/id/E/EX/EXODIST/Test-SharedFork-0.35.tar.gz ... OK
Configuring Test-SharedFork-0.35 ... OK
Building and testing Test-SharedFork-0.35 ... OK
Successfully installed Test-SharedFork-0.35
<== Installed dependencies for .. Finishing.
1 distribution installed
C:\Users\cochrane\Test-TCP>perl Makefile.PL
Generating a dmake-style Makefile
Writing Makefile for Test::TCP
Writing MYMETA.yml and MYMETA.json
C:\Users\cochrane\Test-TCP>dmake test
cp lib/Test/TCP.pm blib\lib\Test\TCP.pm
cp lib/Test/TCP/CheckPort.pm blib\lib\Test\TCP\CheckPort.pm
cp lib/Net/EmptyPort.pm blib\lib\Net\EmptyPort.pm
"C:\Strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
t/00_compile.t ................. 1/1 # Test::More: 1.001014
t/00_compile.t ................. ok
t/01_simple.t .................. ok
t/02_abrt.t .................... skipped: win32 doesn't support embedded functio
n named dump()
t/03_return_when_sigterm.t ..... ok
t/04_die.t ..................... ok
t/05_sigint.t .................. skipped: this test requires SIGUSR1
t/06_nest.t .................... ok
t/07_optional.t ................ ok
t/08_exit.t .................... ok
t/09_fork.t .................... ok
t/10_oo.t ...................... ok
t/11_net_empty_port.t .......... 2/?     # found an empty IPv6 port
t/11_net_empty_port.t .......... ok
t/12_pass_wait_port_options.t .. ok
t/13_undef_port.t .............. ok
t/14_listen.t .................. ok
All tests successful.
Files=15, Tests=144, 98 wallclock secs ( 0.11 usr +  0.09 sys =  0.20 CPU)
Result: PASS
```

It seems that #67 isn't forcing an update of `Test::Simple`, nevertheless it also seems that the problem has been fixed.

**Comment by nponeccop** [comment]
> It seems that #67 isn't forcing an update of `Test::Simple`

It's very strange. Can you check that Makefile.PL contains   `"Test::More" => "1.302136"`?

If `Test::TCP` works with stock `Test-Simple` then #67 is pointless. I'll investigate the situation.

**Comment by paultcochrane** [comment]
Here's the generated `Makefile.PL` from SP 5.20.3.1 so that you have the full details:

```
# =========================================================================
# THIS FILE IS AUTOMATICALLY GENERATED BY MINILLA.
# DO NOT EDIT DIRECTLY.
# =========================================================================

use 5.006;
use strict;

use ExtUtils::MakeMaker 6.64;


my %WriteMakefileArgs = (
    NAME     => 'Test::TCP',
    DISTNAME => 'Test-TCP',
    VERSION  => '2.19',
    EXE_FILES => [glob('script/*'), glob('bin/*')],
    CONFIGURE_REQUIRES => {
  "ExtUtils::MakeMaker" => "6.64"
}
,
    BUILD_REQUIRES     => {}
,
    TEST_REQUIRES      => {
  "File::Temp" => 0,
  "Socket" => 0,
  "Test::More" => "0.98"
}
,
    PREREQ_PM          => {
  "IO::Socket::INET" => 0,
  "IO::Socket::IP" => 0,
  "Test::More" => 0,
  "Test::SharedFork" => "0.29",
  "Time::HiRes" => 0,
  "perl" => "5.008001"
}
,
);

WriteMakefile(%WriteMakefileArgs);
```

I checked out tag 2.14 and it seems I can reproduce the error:

```
C:\Users\cochrane\Test-TCP>dmake test
cp lib/Test/TCP.pm blib\lib\Test\TCP.pm
cp lib/Net/EmptyPort.pm blib\lib\Net\EmptyPort.pm
cp lib/Test/TCP/CheckPort.pm blib\lib\Test\TCP\CheckPort.pm
"C:\Strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-
e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/
*.t
t/00_compile.t ................. 1/1 # Test::More: 1.001014
t/00_compile.t ................. ok
t/01_simple.t .................. 1/? cannot open port: ::1:50257 at C:\Users\coc
hrane\Test-TCP\blib\lib/Test/TCP.pm line 54.
    # Child (v6) exited without calling finalize()
t/01_simple.t .................. 2/?
#   Failed test 'v6'
#   at C:/Strawberry/perl/lib/Test/Builder.pm line 279.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 2.
t/01_simple.t .................. Dubious, test returned 255 (wstat 65280, 0xff00
)
Failed 1/2 subtests
t/02_abrt.t .................... skipped: win32 doesn't support embedded functio
n named dump()
t/03_return_when_sigterm.t ..... ok
t/04_die.t ..................... ok
t/05_sigint.t .................. skipped: this test requires SIGUSR1
t/06_nest.t .................... ok
t/07_optional.t ................ ok
t/08_exit.t .................... ok
t/09_fork.t .................... ok
t/10_oo.t ...................... ok
t/11_net_empty_port.t .......... 3/?
    #   Failed test 'port is open'
    #   at t/11_net_empty_port.t line 22.
    # Looks like you failed 1 test of 4.

t/11_net_empty_port.t .......... 4/? #   Failed test 'v6'
#   at t/11_net_empty_port.t line 37.
# Looks like you failed 1 test of 4.
t/11_net_empty_port.t .......... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/4 subtests
t/12_pass_wait_port_options.t .. ok
t/13_undef_port.t .............. ok

Test Summary Report
-------------------
t/01_simple.t                (Wstat: 65280 Tests: 2 Failed: 1)
  Failed test:  2
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
t/11_net_empty_port.t        (Wstat: 256 Tests: 4 Failed: 1)
  Failed test:  4
  Non-zero exit status: 1
Files=14, Tests=117, 90 wallclock secs ( 0.06 usr +  0.13 sys =  0.19 CPU)
Result: FAIL
Failed 2/14 test programs. 2/117 subtests failed.
dmake:  Error code 255, while making 'test_dynamic'
```

... although it turns up in `01_simple.t` now and in my initial post it was in `00_compile.t`.

Bisecting the issue showed that c93001217fa69ccc786fedb210505a4ae7e5be23 was the first commit where the problem was resolved.

Hope this helps!

**Comment by nponeccop** [comment]
> Here's the generated Makefile.PL from SP 5.20.3.1 so that you have the full details

Are you sure it's from #67? It seems it's not. Anyway, it's not important now as the bumping turned out to be useless. #68 shows that everything works fine now and upgrading is not necessary, so it changes the CI script not to do it.

**Comment by paultcochrane** [comment]
Sorry, my bad.  I thought (for some reason) that #67 had already been merged, hence I tested against master.  Dunno what I was thinking really :-/  Oh well, at least this issue seems to be resolved.

**Comment by nponeccop** [comment]
This was released as Test-TCP-1.20. Unfortunately there is a regression in Test-TCP-1.21: #84 so use 1.20 for now.
